datafusion/core/tests/sql/aggregates.rs

```diff
@@ -37,7 +37,7 @@ async fn csv_query_array_agg_distinct() -> Result<()> {
         Schema::new(vec![Field::new_list(
             "ARRAY_AGG(DISTINCT aggregate_test_100.c2)",
             Field::new("item", DataType::UInt32, true),
-            false
+            true
```

Comment:
I think `nullable` should be fixed together otherwise the test seems like regression to me 🤔

Reply:
I noticed that we need further changes to resolve this one. See my comment here: https://github.com/apache/datafusion/pull/11063